hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -1527,6 +1652,278 @@ public Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
 		}
 	}
 
+	protected class AllowancePrecompile extends ERCReadOnlyAbstractPrecompile {
+		private TokenAllowanceWrapper allowanceWrapper;
+
+		public AllowancePrecompile(final TokenID tokenId) {
+			super(tokenId);
+		}
+
+		@Override
+		public TransactionBody.Builder body(final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
+			final var nestedInput = input.slice(24);
+			allowanceWrapper = decoder.decodeTokenAllowance(nestedInput, aliasResolver);
+
+			return super.body(input, aliasResolver);
+		}
+
+		@Override
+		public Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
+			final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger = ledgers.accounts();
+			final var allowance = (TreeMap<FcTokenAllowanceId, Long>) accountsLedger.get(allowanceWrapper.owner(), FUNGIBLE_TOKEN_ALLOWANCES);
+			long value = 0;
+			for (Map.Entry<FcTokenAllowanceId, Long> e : allowance.entrySet()) {
+				if (allowanceWrapper.spender().getAccountNum() == e.getKey().getSpenderNum().longValue()) {
+					value = e.getValue();
+				}
+			}
```

Comment:
You can extract the logic below in a default method in the Precompile interface and reuse it where it's needed (e.g. ApprovePrecompile:

```
default long getFungibleAllowanceValueForAccount(final AccountID owner, final long accountNum) {
    final var allowance = (TreeMap<FcTokenAllowanceId, Long>) accountsLedger.get(allowanceWrapper.owner(), 
         FUNGIBLE_TOKEN_ALLOWANCES);
			long value = 0;
			for (Map.Entry<FcTokenAllowanceId, Long> e : allowance.entrySet()) {
				if (allowanceWrapper.spender().getAccountNum() == 
                           e.getKey().getSpenderNum().longValue()) {
					value = e.getValue();
				}
			}
}
```